ttnn/cpp/ttnn/operations/experimental/ccl/send_recv_async/recv_async/device/recv_async_op_program_factory.cpp

```diff
@@ -301,50 +324,54 @@ tt::tt_metal::operation::ProgramWithCallbacks recv_async_multicore(
         }
     }
 
-    if (!socket_storage_in_dram) {
-        auto override_runtime_arguments_callback =
-            [receiver_core_coords, writer_kernel](
-                const void* operation,
-                tt::tt_metal::Program& program,
-                const std::vector<Tensor>& input_tensors,
-                const std::vector<std::optional<const Tensor>>& optional_input_tensors,
-                const std::vector<Tensor>& output_tensors) {
-                const auto& mesh_socket = static_cast<const ttnn::RecvAsync*>(operation)->mesh_socket;
-
-                for (uint32_t core_idx = 0; core_idx < receiver_core_coords.size(); ++core_idx) {
-                    const auto& receiver_core_coord = receiver_core_coords[core_idx];
-                    auto& writer_runtime_args = GetRuntimeArgs(program, writer_kernel, receiver_core_coord);
-
-                    writer_runtime_args[0] = mesh_socket.get_config_buffer()->address();
-                    writer_runtime_args[1] = input_tensors[0].buffer()->address();
-                }
-            };
+    return {
+        std::move(program),
+        shared_variables_t{
+            .receiver_core_coords = receiver_core_coords,
+            .reader_kernel_id = reader_kernel,
+            .writer_kernel_id = writer_kernel,
+            .socket_storage_in_dram = socket_storage_in_dram,
+        }};
+}
 
-        return {
-            .program = std::move(program), .override_runtime_arguments_callback = override_runtime_arguments_callback};
-    } else {
-        auto override_runtime_arguments_callback =
-            [receiver_core_coords, reader_kernel, writer_kernel](
-                const void* operation,
-                tt::tt_metal::Program& program,
-                const std::vector<Tensor>& input_tensors,
-                const std::vector<std::optional<const Tensor>>& optional_input_tensors,
-                const std::vector<Tensor>& output_tensors) {
-                const auto& mesh_socket = static_cast<const ttnn::RecvAsync*>(operation)->mesh_socket;
-
-                for (uint32_t core_idx = 0; core_idx < receiver_core_coords.size(); ++core_idx) {
-                    const auto& receiver_core_coord = receiver_core_coords[core_idx];
-                    auto& reader_runtime_args = GetRuntimeArgs(program, reader_kernel, receiver_core_coord);
-                    auto& writer_runtime_args = GetRuntimeArgs(program, writer_kernel, receiver_core_coord);
-
-                    reader_runtime_args[0] = mesh_socket.get_config_buffer()->address();
-                    writer_runtime_args[0] = input_tensors[0].buffer()->address();
-                }
-            };
+void RecvAsyncMeshWorkloadFactory::override_runtime_arguments(
+    cached_mesh_workload_t& cached_workload,
+    const operation_attributes_t& operation_attributes,
+    const tensor_args_t& tensor_args,
+    tensor_return_value_t& tensor_return_value) {
```

Comment:
Standardized attribute since C++17. Same for `coordinate_range`.
```suggestion
    [[maybe_unused]] tensor_return_value_t& tensor_return_value) {
```

Reply:
Thanks for the tip. I added the attribute to `tensor_return_value`. I also realize `coordinate_range` is used after all, so removed the `(void)` line associated with it.